lib/src/collection/treeset.dart

```diff
@@ -645,76 +645,83 @@ class AvlTreeSet<V> extends TreeSet<V> {
   /// See [IterableBase.first]
   @override
   V get first {
-    if (_root == null) return null;
-    AvlNode<V> min = _root.minimumNode;
-    return min != null ? min.object : null;
+    _TreeNode<V>? min = _root?.minimumNode;
+    if (min != null) {
+      return min.object;
+    }
+    throw StateError('No first element');
```

Comment:
What if `V` is nullable and the first element is `null`?

Reply:
I'm a little confused - the only null check here is on the Node; which must not be null. The object it carries (V) can be null and is returned at line 650.  

Or are you suggesting the signature should be `V?`

Reply:
My mistake, mixed up the node with the value. SG.

Reply:
There are a lot of assumptions here that preclude supporting nullable values -- perhaps the answer is to set the type to `V extends Object` for now then make everything safe for null objects.